tests/test_nanoleaf_client.py

```
@@ -0,0 +1,286 @@
+        client._stream = lambda target, panels: streamed.append((target, list(panels)))  # type: ignore[method-assign]
+        client.send_color(device, 1.0, 0.0, 0.0, 1.0)
+        self.assertEqual(len(streamed), 1)
+        panel_ids = [row[0] for row in streamed[0][1]]
+        self.assertEqual(panel_ids, [8954, 64823])
+        self.assertNotIn(0, panel_ids)
+
+    def test_stream_drops_when_interval_has_not_elapsed(self):
+        from nanoleaf_client import NanoleafClient, NanoleafDevice, MIN_STREAM_INTERVAL
+        client = NanoleafClient()
+        self.addCleanup(client.close)
+        device = NanoleafDevice('nl_x', '127.0.0.1', auth_token='tok', model='NL42')
```

Ruff (0.16.1)

[error] Possible hardcoded password assigned to argument: "auth_token"

(S106)